flink-libraries/flink-table/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -743,6 +754,14 @@ private static int julianExtract(TimeUnitRange range, int julian) {
 		switch (range) {
 			case YEAR:
 				return year;
+			case ISOYEAR:
```

Comment:
1. Please extract this code into some method
2. add some comments what is going on here
3. Please add more tests for `ISOYEAR` covering those edge cases (when `ISOYEAR` differs from `YEAR` in either of the directions).